### PR TITLE
[FtpUploadV2] Resolve dependabot alert about shelljs

### PR DESCRIPTION
**Pull request by DaniilShmelev**
**Task name**: FtpUploadV2

**Description**: Bumped tasklib to 3.x

**Documentation changes required:** No

**Added unit tests:** No

**Attached related issue:** No

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by DaniilShmelev** [comment]
Tested these changes manually by running canary on windows and debian. Everything works as expected.

File: Tasks/FtpUploadV2/Tests/L0NoClean.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        "*": [
+            "rootFolder/a",
+            "rootFolder/b",
+            "rootFolder/c"
+        ]
+    }
+};
+tr.setAnswers(a);
+
+tr.run();
```
Comment (DaniilShmelev) [inline Tasks/FtpUploadV2/Tests/L0NoClean.ts:39]: @tatyana-kostromskaya could you please fix EOLs in the new tests?

Comment (DaniilShmelev) [inline Tasks/FtpUploadV2/Tests/L0NoClean.ts:39]: @tatyana-kostromskaya could you please fix EOLs in the new tests?

File: Tasks/FtpUploadV2/task.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "demands": [],
     "version": {
         "Major": 2,
-        "Minor": 198,
+        "Minor": 200,
```
Comment (anatolybolshakov) [inline Tasks/FtpUploadV2/task.json:21]: Please bump to 201

File: Tasks/FtpUploadV2/ftpuploadtask.ts [hunk 3]
```diff
@@ -166,22 +166,22 @@ function getFtpOptions(): FtpOptions {
...
         serverEndpointUrl = url.parse(
-            tl.getEndpointUrl(serverEndpoint, false)
+            tl.getEndpointUrl(serverEndpoint, false)!
         );
 
         const serverEndpointAuth: tl.EndpointAuthorization = tl.getEndpointAuthorization(
             serverEndpoint,
             false
-        );
+        )!;
```
Comment (anatolybolshakov) [inline Tasks/FtpUploadV2/ftpuploadtask.ts:177]: Why it can't be falsey value here?

Comment (denis-tikhomirov) [reply Tasks/FtpUploadV2/ftpuploadtask.ts:177]: Next serverEndpointAuth is used to get username and password in the code. And the fields are marked as required in case of entering creds manually (not using of service connection). So IMO it's correct to require declaring of username and password here.

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

File: Tasks/FtpUploadV2/Tests/L0.ts [hunk 4]
```diff
@@ -1,16 +1,129 @@
...
+
+    it('check args: no files found', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NoFilesFound.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('loc_mock_NoFilesFound'), 'Should have printed: loc_mock_NoFilesFound');
+        assert(tr.failed, 'task should have failed');
+
```
Comment (DaniilShmelev) [inline Tasks/FtpUploadV2/Tests/L0.ts:126]: We don't have a single unit test for successful execution case. Maybe we should add them? @tatyana-kostromskaya

**Review by DaniilShmelev — COMMENTED** [review]
LGTM, but take a look at the comment

File: Tasks/FtpUploadV2/Tests/L0.ts (hunk 4 above)
Comment (anatolybolshakov) [reply Tasks/FtpUploadV2/Tests/L0.ts:126]: Added test, thanks!

**Comment by DaniilShmelev** [comment]
LGTM

File: Tasks/FtpUploadV2/Tests/package-lock.json [hunk 5]
```diff
@@ -1,8 +1,24 @@
 {
   "name": "ftp-upload-tests",
   "version": "1.0.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (DaniilShmelev) [inline Tasks/FtpUploadV2/Tests/package-lock.json:4]: @anatolybolshakov maybe we should revert this change to the `lockfileVersion`

Comment (anatolybolshakov) [reply Tasks/FtpUploadV2/Tests/package-lock.json:4]: @DaniilShmelev as I know v2 version is backward compatible to previous versions of npm - so there should not be issues

**Review by anatolybolshakov — APPROVED** [review]
Approved from behalf of @DaniilShmelev (since I did last changes, but author can't approve his own PRs)
